drivers/gpio/Kconfig.gd32

```diff
@@ -5,5 +5,6 @@ config GPIO_GD32
 	bool "GD32 GPIO driver"
 	default y
 	depends on DT_HAS_GD_GD32_GPIO_ENABLED
+	imply GD32_EXTI
```

Comment:
Should this be enabled by default?

Reply:
Yes, it is.
Interrupts handling needs this option.
The callback function of the GPIO API will not work If this option is disabled.
This PR is intended mainly to clarify dependencies.

Reply:
If it's required, should be `depends on`. `imply` means that it is optional but should be the default

Reply:
I added more change gpio_add_callback/gpio_remove_callback api returns -ENOTSUP if GD32_EXTI disabled.

It make allow to disable GD32_EXTI without API contradiction.

https://github.com/zephyrproject-rtos/zephyr/pull/50879/files#diff-606da3fe7a304f5544526be564097f77dc7db8e627c004fc62b8eed50e1b45dbR335-R341